base-notebook/start.sh

```diff
@@ -239,13 +236,13 @@ else
     # A misconfiguration occurs when the user modifies the default values of
     # NB_USER, NB_UID, or NB_GID, but we cannot update those values because we
     # are not root.
-    if [[ "${NB_USER}" != "jovyan" && "${NB_USER}" != "$(id -un)" ]]; then
+    if [[ "${NB_USER}" != "$(id -un)" && "${NB_USER}" != "jovyan" ]]; then
         _log "WARNING: container must be started as root to change the desired user's name with NB_USER=\"${NB_USER}\"!"
     fi
-    if [[ "${NB_UID}" != "${JOVYAN_UID}" && "${NB_UID}" != "$(id -u)" ]]; then
+    if [[ "${NB_UID}" != "$(id -u)" && "${NB_UID}" != "$(id -u jovyan)" ]]; then
```

Comment:
I changed the order of these two expressions.
Now, it won't execute `id -u jovyan` if the NB_UID is the one the current user has.
And because we don't call this expression, this will work fine when there is no `jovyan` user at all.